### PR TITLE
Add support for right click human actions

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

Title

@Walarks this includes item attacks

## Changelog
:cl:
code: Add support for right click human actions
/:cl:

### CODE REVIEW

File: code/_onclick/item_attack.dm [hunk 1]
```diff
@@ -187,3 +189,136 @@
...
+ */
+/obj/item/proc/melee_attack_chain_alternate(mob/user, atom/target, params)
+	// Return TRUE in attackby_alternate() to prevent afterattack() effects (when safely moving items for example)
+	var/resolved = target.attackby_alternate(src, user, params)
+	if(resolved || QDELETED(target) || QDELETED(src))
+		return
+	afterattack_alternate(target, user, TRUE, params) // TRUE: clicking something Adjacent
+
+/atom/proc/attackby_alternate(obj/item/I, mob/user, params)
+	SIGNAL_HANDLER_DOES_SLEEP
```
Comment (SplinterGP) [inline code/_onclick/item_attack.dm:217]: bad

Comment (spookydonut) [reply code/_onclick/item_attack.dm:217]: aaaaaaaaaaaaaaaaaaaaa

**Review by spookydonut — COMMENTED** [review]
ci failure

File: code/_onclick/click.dm [hunk 2]
```diff
@@ -143,17 +143,14 @@
...
 			UnarmedAttack(A, TRUE, modifiers)
 	else
 		if(W)
-			var/attack
 			var/proximity = A.Adjacent(src)
-			if(proximity && A.attackby(W, src, params))
-				attack = TRUE
-			if(!attack)
-				W.afterattack(A, src, proximity, params)
+			if(proximity)
```
Comment (SplinterGP) [inline code/_onclick/click.dm:152]: remove var above
```suggestion
			if(A.Adjacent(src))
```

**Review by SplinterGP — CHANGES_REQUESTED** [review]

File: code/_onclick/click.dm [hunk 3]
```diff
@@ -143,17 +143,14 @@
...
 	if(CanReach(A, W))
 		if(W)
-			W.melee_attack_chain(src, A, params)
+			W.melee_attack_chain(src, A, params, modifiers["right"])
 		else
 			UnarmedAttack(A, TRUE, modifiers)
 	else
 		if(W)
-			var/attack
 			var/proximity = A.Adjacent(src)
```
Comment (TiviPlus) [inline code/_onclick/click.dm:151]:
```suggestion
```

**Pull request merged**